rz-cmn: Support build meta-reneas for rz-cmn MACHINE

khanhduydo-rvc opened the pull request:
- Modify linux-yocto recipes for supporting multi boards
- Modify flash-writer, ATF and u-boot recipes for rz-cmn MACHINE
- Fix build errors when adapting new MACHINE

khanhduydo-rvc committed:
rz-cmn: Modify recipes-kernel folder

- Add build support for rz-cmn MACHINE
- Modify linux-yocto recipes to support 4 boards: rzg2l-sbc, rzg2l-evk, rzv2l-evk, rzv2h-evk

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rz-cmn: Modify recipes-bsp folder

- Add build support for flash writer, ATF, and U-Boot for 4 boards: rzg2l-sbc, rzg2l-evk, rzv2l-evk, rzv2h-evk

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rz-cmn: Modify remaining folders

- Add build support for rz-cmn MACHINE
- Modify Readme to adapt to new MACHINE

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

vudangRVC commented on conf/machine/rz-cmn.conf:
```diff
@@ -1,6 +1,6 @@
+#@NAME: RZ-CMN-yocto machine
+#@DESCRIPTION: Reference machine configuration for Renesas RZ Common boards
 
 SOC_FAMILY = "r9a07g044l"
```
@khanhduydo-rvc 
Do you still use this variable?

khanhduydo-rvc replied:
In fact, this variable is not used anywhere in meta-renesas  so we can remove it if we want

vudangRVC commented on include/core-image-renesas-base.inc:
```diff
@@ -30,22 +30,26 @@ IMAGE_ROOTFS_EXTRA_SPACE = "1048576"
 IMAGE_ROOTFS_ALIGNMENT = "16"
 
 # Move images to boot partition
+<<<<<<< HEAD
```
What is this ??? Have you verified the changes?

khanhduydo-rvc replied:
my mistake when using cherry-pick. Fixed !

tiennguyenuh commented on recipes-bsp/trusted-firmware-a/trusted-firmware-a.bb:
```diff
@@ -10,20 +10,25 @@ LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda
 
-SRC_URI:rzg2l-sbc = " \
-    git://github.com/Renesas-SST/rz-atf.git;name=machine;branch=${BRANCH};protocol=https \
+SRC_URI:rz-cmn = " \
+    git://github.com/vudangRVC/rz-atf-sst.git;name=rzg2l-sbc;subdir=rzg2l-sbc;branch=atf-pass-params;protocol=https \
+    git://github.com/vudangRVC/rz-atf-sst.git;name=rzg2l-evk;subdir=rzg2l-evk;branch=atf-pass-params-g2l;protocol=https \
+    git://github.com/vudangRVC/rz-atf-sst.git;name=rzv2l-evk;subdir=rzv2l-evk;branch=atf-pass-params-v2l;protocol=https \
+    git://github.com/vudangRVC/rz-atf-sst.git;name=rzv2h-evk;subdir=rzv2h-evk;branch=atf-pass-params-v2h;protocol=https \
```
Make sure to point to the SST repo once all the boards are integrated into a single branch. The target branch should be `styhead/rz-cmn"

tiennguyenuh commented on recipes-bsp/u-boot/u-boot_2021.10.bb:
```diff
@@ -7,10 +7,10 @@ DEPENDS += "bc-native dtc-native"
-SRC_URI:rzg2l-sbc = "${UBOOT_URL};name=machine;protocol=https;branch=${BRANCH}"
-BRANCH:rzg2l-sbc = "styhead/rz-sbc"
-SRCREV_machine:rzg2l-sbc = "${AUTOREV}"
+UBOOT_URL = "git://github.com/vudangRVC/u-boot-sst.git"
```
Make sure to point to the SST repo once all the boards are integrated into a single branch. The target branch should be `styhead/rz-cmn`

tiennguyenuh commented on recipes-kernel/linux/README.md:
Can you update this README based on the new structure of the linux folder?

tiennguyenuh commented on recipes-kernel/linux/rz-cmn/rzv2h-evk/0002-rzv2h-evk-Support-RZV2H-EVK.patch:
The PR to support RZ/V2H-EVK is currently under review. Make sure to update the patch once the PR is merged

tiennguyenuh replied:
I just want to make sure the patch is up to date. The PR in the Linux repository has been merged recently (https://github.com/Renesas-SST/linux-rz/pull/8), and I believe there were some changes made after the review.

tiennguyenuh commented on recipes-bsp/firmware-pack/firmware-pack.bb:
```diff
@@ -7,45 +7,64 @@ inherit deploy
 
 S = "${WORKDIR}/sources"
 
+TARGETS = "rzg2l-sbc rzg2l-evk rzv2l-evk rzv2h-evk"
```
Find a better approach to define supported board.

tiennguyenuh commented on recipes-bsp/flash-writer/flash-writer.bb:
```diff
@@ -11,58 +11,61 @@ ENABLE_RZ_SCE		?= '0'
+UNPACKDIR = "${S}"
 
-	oe_runmake BOARD=${BOARD}
+TARGETS = "rzg2l-sbc rzg2l-evk rzv2l-evk"
```
Same

tiennguyenuh commented on recipes-bsp/trusted-firmware-a/trusted-firmware-a.bb:
```diff
@@ -51,30 +56,65 @@ LDFLAGS[unexport] = "1"
+    make ${BUILD_FLAGS} bl2 bl31
+
+    cd ${S}/rzv2l-evk
+    BUILD_FLAGS="PLAT=v2l BOARD=smarc_pmic_2"
```
For V2L-EVK, change the BOARD to `smarc_rzv2l` instead of smarc_pmic_2.

khanhduydo-rvc replied:
I don't see smarc_rzv2l board option. It leads to this error: 
`plat/renesas/rz/soc/v2l/platform.mk:8: plat/renesas/rz/board/smarc_rzv2l/rz_board.mk: No such file or directory`

tiennguyenuh replied:
Actually, the new board has been merged into the `styhead/rz-cmn` branch: https://github.com/Renesas-SST/rz-atf/blob/styhead/rz-cmn/plat/renesas/rz/board/smarc_rzv2l/rz_board.mk

tiennguyenuh commented on include/core-image-renesas-base.inc:
```diff
@@ -55,13 +54,13 @@ do_release_clean() {
     rm -f ${DEPLOY_DIR_IMAGE}/*initramfs*
     rm -f ${DEPLOY_DIR_IMAGE}/Image*
-    rm -f ${DEPLOY_DIR_IMAGE}/rzg2l-sbc*
+    rm -f ${DEPLOY_DIR_IMAGE}/*.dtb
```
I think you missed cleaning the `*.dtbo` as well.

tiennguyenuh commented on recipes-bsp/flash-writer/flash-writer.bb:
```diff
@@ -11,58 +11,61 @@ ENABLE_RZ_SCE		?= '0'
 
 FLASH_WRITER_URL = "git://github.com/Renesas-SST/flash-writer.git"
 BRANCH = "dunfell/rz-sbc"
```
It should point to styhead/rz-cmn branch.

tiennguyenuh commented on recipes-bsp/trusted-firmware-a/trusted-firmware-a.bb:
```diff
@@ -51,30 +56,65 @@ LDFLAGS[unexport] = "1"
+do_compile() {
+    cd ${S}/rzg2l-sbc
+    BUILD_FLAGS="PLAT=g2l BOARD=sbc_1"
+    make ${BUILD_FLAGS} bl2 bl31
```
For new FCONF integrated into the TF-A repository, we should build with the `bl2_with_dtb` option instead of `bl2`. The same applies to other boards (g2l-evk, v2l and v2h).

To be able to build this change, you'll need dtc-native as a dependency. Please add it to the recipe like this.

```
DEPENDS:append = " dtc-native"
```

khanhduydo-rvc committed:
rz-cmn: Update new machanism for support targets

- Update new machanism for support targets for trusted-firmware-a, flash-writer and firmware-pack recipes.

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

vudangRVC commented on recipes-kernel/linux/linux-yocto_5.10.bb:
@khanhduydo-rvc , we don't support "rz-cmn" on Dunfell. Please revert the change.

khanhduydo-rvc committed:
rz-cmn: Update uEnv.txt file

- Update uEnv.txt file to use board_id variable from U-boot

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

tiennguyenuh reviewed (commented):
Some target images in the dynamic layers need to be modified. I'm still seeing that they include `require include/core-image-renesas-sbc.inc`. Although we are not currently testing these images, but the build will fail if the meta-layer is included in bblayers

khanhduydo-rvc committed:
rz-cmn: Update Trusted Firmware A repo URL

- Change repo URL from personal repo to global repo.

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

vudangRVC commented on recipes-bsp/flash-writer/flash-writer.bb:
```diff
@@ -10,59 +10,65 @@ ENABLE_SPD_OPTEE	?= '0'
-SRC_URI = "${FLASH_WRITER_URL};protocol=https;branch=${BRANCH}"
-SRCREV = "${AUTOREV}"
+BRANCH = "styhead/rz-cmn"
+SRC_URI = "${FLASH_WRITER_URL};name=rzg2l-sbc;subdir=rzg2l-sbc;protocol=https;branch=${BRANCH} \
```
Can you explain this configuration?

vudangRVC commented on recipes-bsp/flash-writer/flash-writer.bb:
```diff
@@ -10,59 +10,65 @@ ENABLE_SPD_OPTEE	?= '0'
+"
+SRCREV_rzg2l-sbc = "8e5919a314673217d93dbb34227b8c22d71d681b"
+SRCREV_rzg2l-evk = "ff167b676547f3997906c82c9be504eb5cff8ef0"
+SRCREV_rzv2l-evk = "ff167b676547f3997906c82c9be504eb5cff8ef0"
```
Why do we need many SRCREV variables?

vudangRVC commented on recipes-bsp/flash-writer/flash-writer.bb:
```diff
@@ -10,59 +10,65 @@ ENABLE_SPD_OPTEE	?= '0'
+SRCREV_rzg2l-evk = "ff167b676547f3997906c82c9be504eb5cff8ef0"
+SRCREV_rzv2l-evk = "ff167b676547f3997906c82c9be504eb5cff8ef0"
+
+SRCREV_FORMAT = "rzg2l-sbc_rzg2l-evk_rzv2l-evk"
```
Explain this variable?

vudangRVC commented on recipes-bsp/trusted-firmware-a/trusted-firmware-a.bb:
```diff
@@ -10,20 +10,27 @@ LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda
-    git://github.com/Renesas-SST/rz-atf.git;name=machine;branch=${BRANCH};protocol=https \
+DEPENDS:append = " dtc-native xxd-native"
+
+SRC_URI:rz-cmn = " \
```
The same question as the ATF

tiennguyenuh commented on recipes-bsp/trusted-firmware-a/trusted-firmware-a.bb:
```diff
@@ -51,30 +58,66 @@ LDFLAGS[unexport] = "1"
+
+        install -m 0644 ${D}/boot/bl2-${target}.bin ${DEPLOYDIR}/target/images/bl2-${target}.bin
+    done
 }
```
Please use the SUPPORT_TARGETS variable that we have already defined in local.conf.

khanhduydo-rvc committed:
rz-cmn: add do_prepare_src task

- Add do_prepare_src task to create a build directory for each board so
we can simplify the SRC_URI variable for do_fetch task.

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rz-cmn: Change include files from rzg2l to rz-cmn

- Change include files from rzg2l to rz-cmn on QT5 and QT6 components

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc commented:
> Some target images in the dynamic layers need to be modified. I'm still seeing that they include `require include/core-image-renesas-sbc.inc`. Although we are not currently testing these images, but the build will fail if the meta-layer is included in bblayers

disable qt5 and qt6 components will fix it

vudangRVC commented on recipes-bsp/trusted-firmware-a/trusted-firmware-a.bb:
```diff
@@ -10,20 +10,20 @@ LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda
-FLASH_ADDRESS_FIP:rzg2l-sbc = "1D200"
+# Configuration for multi boards
+FLASH_ADDRESS_BL2_BP:rz-cmn = "00000"
+FLASH_ADDRESS_FIP:rz-cmn = "1D200"
```
RZV2H has different flashing addresses. I approve for now. Please update in the next PR
